src/Microsoft.DotNet.ProjectModel/Files/IncludeFilesResolver.cs

```diff
@@ -74,28 +74,68 @@ public static IEnumerable<IncludeEntry> GetIncludeFiles(IncludeContext context,
                 }
                 else if (isFile && files.Any())
                 {
-                    includeEntries.Add(new IncludeEntry(targetBasePath, files.First()));
+                    includeEntries.Add(
+                        new IncludeEntry(
+                            targetBasePath,
+                            Path.GetFullPath(
+                                Path.Combine(sourceBasePath, PathUtility.GetPathWithDirectorySeparator(files.First().Path)))));
                 }
-                else
+                else if (!isFile)
                 {
                     targetBasePath = targetBasePath.Substring(0, targetBasePath.Length - 1);
 
                     foreach (var file in files)
                     {
+                        var fullPath = Path.GetFullPath(
+                            Path.Combine(sourceBasePath, PathUtility.GetPathWithDirectorySeparator(file.Path)));
                         string targetPath = null;
 
                         if (flatten)
```

Comment:
`true` for `flatten` now means it is in `mappings` section https://github.com/dotnet/cli/blob/9fc75dd9fc7556399842df5bb885bdd02902f423/src/Microsoft.DotNet.ProjectModel/Files/IncludeFilesResolver.cs#L148